staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers_test.go

```diff
@@ -0,0 +1,303 @@
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+)
+
+func TestSerializeObject(t *testing.T) {
+	smallPayload := []byte("{test-object,test-object}")
+	largePayload := bytes.Repeat([]byte("0123456789abcdef"), defaultGzipThresholdBytes/16+1)
+	tests := []struct {
+		name string
+
+		compressionEnabled bool
+
+		mediaType  string
+		out        []byte
+		outErrs    []error
+		req        *http.Request
+		statusCode int
+		object     runtime.Object
+
+		wantCode    int
+		wantHeaders http.Header
+		wantBody    []byte
+	}{
+		{
+			name:        "serialize object",
+			out:         smallPayload,
+			req:         &http.Request{Header: http.Header{}},
+			wantCode:    http.StatusOK,
+			wantHeaders: http.Header{"Content-Type": []string{""}},
+			wantBody:    smallPayload,
+		},
+
+		{
+			name:        "return content type",
+			out:         smallPayload,
+			mediaType:   "application/json",
+			req:         &http.Request{Header: http.Header{}},
+			wantCode:    http.StatusOK,
+			wantHeaders: http.Header{"Content-Type": []string{"application/json"}},
+			wantBody:    smallPayload,
+		},
+
+		{
+			name:        "return status code",
+			statusCode:  http.StatusBadRequest,
+			out:         smallPayload,
+			mediaType:   "application/json",
+			req:         &http.Request{Header: http.Header{}},
+			wantCode:    http.StatusBadRequest,
+			wantHeaders: http.Header{"Content-Type": []string{"application/json"}},
+			wantBody:    smallPayload,
+		},
+
+		{
+			name:        "fail to encode object",
+			out:         smallPayload,
+			outErrs:     []error{fmt.Errorf("bad")},
+			mediaType:   "application/json",
+			req:         &http.Request{Header: http.Header{}},
+			wantCode:    http.StatusInternalServerError,
+			wantHeaders: http.Header{"Content-Type": []string{"application/json"}},
+			wantBody:    smallPayload,
+		},
+
+		{
+			name:        "fail to encode object or status",
+			out:         smallPayload,
+			outErrs:     []error{fmt.Errorf("bad"), fmt.Errorf("bad2")},
+			mediaType:   "application/json",
+			req:         &http.Request{Header: http.Header{}},
+			wantCode:    http.StatusInternalServerError,
+			wantHeaders: http.Header{"Content-Type": []string{"text/plain"}},
+			wantBody:    []byte(": bad"),
+		},
+
+		{
+			name:        "fail to encode object or status with status code",
+			out:         smallPayload,
+			outErrs:     []error{errors.NewNotFound(schema.GroupResource{}, "test"), fmt.Errorf("bad2")},
+			mediaType:   "application/json",
+			req:         &http.Request{Header: http.Header{}},
+			statusCode:  http.StatusOK,
+			wantCode:    http.StatusNotFound,
+			wantHeaders: http.Header{"Content-Type": []string{"text/plain"}},
+			wantBody:    []byte("NotFound:  \"test\" not found"),
```

Comment:
Feel free to ignore, but it seems we have some additional whitespace somewhere in the stack...

Reply:
Yeah, I actually was testing the exact output.  The two spaces separates an empty resource string (same message as before this PR).